src/support/archive.cpp

```diff
@@ -193,7 +193,7 @@ Archive::child_iterator Archive::child_end() const { return Child(); }
 namespace {
 struct Symbol {
   uint32_t symbolIndex;
-  uint32_t stringIndex;
+  ptrdiff_t stringIndex;
```

Comment:
Actually looking at the code, this use of stringIndex/strchr is straight-up wrong. It doesn't matter right now because we don't use archives with symbol tables yet. You can leave this as-is (it will warn, as it should... because it's wrong :))